### ISSUE TITLE
[BUG]: Cannot build with Python 3.11.2 since enum has no `_singles_mask_` attribute.

### ISSUE THREAD

**Issue by hpkfft** [wjakob/nanobind#726]
### Problem description

Using the master branch of nanobind on a Debian 12.7 (bookworm) system, I get:
```
FAILED: tests/test_enum_ext.pyi
terminate called after throwing an instance of 'nanobind::python_error'
  what():  Traceback (most recent call last):
  File "/home/paul/master/nanobind/src/stubgen.py", line 1422, in <module>
  File "/home/paul/master/nanobind/src/stubgen.py", line 1355, in main
  File "/usr/lib/python3.11/importlib/__init__.py", line 126, in import_module
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 676, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 573, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 1233, in create_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/usr/lib/python3.11/enum.py", line 789, in __getattr__
AttributeError: _singles_mask_
```
Note that the attribute appears to be missing:
```
$ grep singles /usr/lib/python3.11/enum.py
$
```

### Reproducible example code

```text
CXX=clang++-16 cmake -S . -B build -G Ninja -DCMAKE_BUILD_TYPE=Debug
cd build
ninja

Note: The use of ninja above is not relevant.   Same issue occurs using the default, `make`.
```

**Comment by wjakob** [comment]
cc @nicholasjng

**Comment by nicholasjng** [comment]
Looks like the `_singles_mask_` attribute was added in commit https://github.com/python/cpython/commit/804a973d8e5fd2728aeef268f3e577d2c0b05baa ,which was not released until 3.11.4. I'll prepare a fix.

**Comment by wjakob** [comment]
Closed via #727.
